library/src/main/java/com/novoda/downloadmanager/lib/BatchQuery.java

```diff
@@ -0,0 +1,244 @@
+package com.novoda.downloadmanager.lib;
+
+import android.support.annotation.IntDef;
+import android.support.annotation.NonNull;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class used to query the batches table. Use {@link BatchQuery.Builder} in order to build it.
+ */
+public class BatchQuery {
+    public static final BatchQuery ALL = new BatchQuery(null, null, null);
+    private static final int LOW_END_FAILED_STATUS_CODE = 400;
+    private static final int HIGH_END_FAILED_STATUS_CODE = 600;
+
+    private final String selection;
+    private final String sortOrder;
+    private final String[] selectionArguments;
+
+    BatchQuery(String selection, String sortOrder, String[] selectionArguments) {
```

Comment:
oh, I see, similar to Criteria, then swap there as well and have `sortOrder` first at both places.